packages/hyperion-autologging/src/MainPageUrl.ts

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates. All Rights Reserved.
+ */
+
+'use strict';
+
+import { getLogger } from "@hyperion/hyperion-global";
+import { Hook } from "@hyperion/hyperion-hook";
+
+let mainPageUrl: URL = new URL('http://undefined');
+
+export function getCurrMainPageUrl(): URL {
+  return mainPageUrl;
+}
+export const onMainPageUrlChange: Hook<(url: URL) => void> = new Hook();
+
+export function updateMainPageUrl(url: string) {
+  try {
+    // We need to ensure the url is absolute
+    const newUrl = new URL(
+      url,
+      mainPageUrl.href, // to handle missing cases
+    );
+
+    if (mainPageUrl.href !== newUrl.href) {
+      mainPageUrl = newUrl;
+      onMainPageUrlChange.call(mainPageUrl);
+    } else {
+      getLogger().warn?.(`ignoring url: path didn't change in ${url}`);
+    }
+  } catch (e) {
+    getLogger().error('invalid url');
+  }
+}
+
+// declare var window: Window & {
+//   location: Location,
+// };
+
+// Lets make sure we are running on the main thread in the browser
+if (
+  typeof window === 'object' &&
+  typeof window.location === 'object' &&
+  typeof window.location.href === 'string'
+) {
+  // We are on the main thread and so safe to update automatically
+  // eslint-disable-next-line no-restricted-properties
+  updateMainPageUrl(window.location.href);
```

Comment:
Maybe I'm missing it,  but how is this continually updated?

Reply:
Yes, that's a complicated thing that I was going to address separately. Let me know how far I can push it for this commit.

Reply:
Added some basic supports that handles majority of use cases (especially with latest Chrome) and will enhance it later more.